From: Shane Michael Matthews <shane.matthews@intel.com>
Date: Tue, 1 Feb 2011 11:31:55 -0500
Subject: NVMe: Disable the device before we write the admin queues
Git-commit: 5911f20039ce59d7e7834f0c42151cf759b6f786
References: FATE#313627
Patch-Mainline: v3.7-rc1

In case the card has been left in a partially-configured state,
write 0 to the Enable bit.

Signed-off-by: Shane Michael Matthews <shane.matthews@intel.com>
Signed-off-by: Matthew Wilcox <matthew.r.wilcox@intel.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/block/nvme.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/block/nvme.c b/drivers/block/nvme.c
index bda9117..e3d9215 100644
--- a/drivers/block/nvme.c
+++ b/drivers/block/nvme.c
@@ -668,6 +668,7 @@ static int __devinit nvme_configure_admin_queue(struct nvme_dev *dev)
 	dev->ctrl_config |= (PAGE_SHIFT - 12) << NVME_CC_MPS_SHIFT;
 	dev->ctrl_config |= NVME_CC_ARB_RR | NVME_CC_SHN_NONE;
 
+	writel(0, &dev->bar->cc);
 	writel(aqa, &dev->bar->aqa);
 	writeq(nvmeq->sq_dma_addr, &dev->bar->asq);
 	writeq(nvmeq->cq_dma_addr, &dev->bar->acq);
-- 
1.7.4.2

